# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1420671737 18000
#      Wed Jan 07 18:02:17 2015 -0500
# Node ID b9cb664404b902e0b9cf288f5df4dd5685f64c4a
# Parent  04d6d592c61423eeb21a10d158578904b150a659
avoid indexing before beginning of string

* load-path.cc (load_path::do_find_dir): Avoid indexing before
beginning of string.

diff --git a/libinterp/corefcn/load-path.cc b/libinterp/corefcn/load-path.cc
--- a/libinterp/corefcn/load-path.cc
+++ b/libinterp/corefcn/load-path.cc
@@ -1378,17 +1378,17 @@ load_path::do_find_dir (const std::strin
           if (dname.substr (dname_len - 1) == file_ops::dir_sep_str ())
             {
               dname = dname.substr (0, dname_len - 1);
               dname_len--;
             }
 
           size_t dir_len = dir.length ();
 
-          if (dname_len >= dir_len
+          if (dname_len > dir_len
               && file_ops::is_dir_sep (dname[dname_len - dir_len - 1])
               && dir.compare (dname.substr (dname_len - dir_len)) == 0)
             {
               file_stat fs (p->dir_name);
 
               if (fs.exists () && fs.is_dir ())
                 return p->dir_name;
             }
